Bluepill-runner/Bluepill-runner/BPPacker.m

```diff
@@ -32,34 +31,28 @@ + (NSMutableArray *)packTests:(NSArray *)xcTestFiles withNoSplitList:(NSArray *)
         }
         return NULL;
     }
+    NSMutableDictionary *testsToRunByTestFilePath = [[NSMutableDictionary alloc] init];
+    NSUInteger totalTests = 0;
     for (BPXCTestFile *xctFile in sortedXCTestFiles) {
         if (![noSplit containsObject:[xctFile name]]) {
-            totalTests += [xctFile numTests];
+            NSMutableSet *testsToRun = [[NSMutableSet alloc] initWithArray:[xctFile allTestCases]];
+            if (testCasesToRun) {
+                [testsToRun intersectSet:[[NSSet alloc] initWithArray:testCasesToRun]];
```

Comment:
is this `intersectSet` just for sanity check to make sure the tests are in the .xctest? It is confusing to have `testsToRun` and `testCasesToRun ` side by side